src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -95,33 +99,57 @@ public NeuralSparseQueryBuilder(StreamInput in) throws IOException {
             Map<String, Float> queryTokens = in.readMap(StreamInput::readString, StreamInput::readFloat);
             this.queryTokensSupplier = () -> queryTokens;
         }
+        if (NeuralSparseTwoPhaseParameters.isClusterOnOrAfterMinReqVersionForTwoPhaseSearchSupport()) {
+            this.neuralSparseTwoPhaseParameters = in.readOptionalWriteable(NeuralSparseTwoPhaseParameters::new);
+        }
+        // to be backward compatible with previous version, we need to use writeString/readString API instead of optionalString API
+        // after supporting query by tokens, queryText and modelId can be null. here we write an empty String instead
+        if (StringUtils.EMPTY.equals(this.queryText)) {
+            this.queryText = null;
+        }
+        if (StringUtils.EMPTY.equals(this.modelId)) {
+            this.modelId = null;
+        }
     }
 
     @Override
     protected void doWriteTo(StreamOutput out) throws IOException {
-        out.writeString(fieldName);
-        out.writeString(queryText);
```

Comment:
This will cause backward incompatibility, as are removing things from what has been getting written down during serialization.